### PR TITLE
refactor(next/dev): reusable start_server

### PR DESCRIPTION
**Pull request by kwonoj**
Extract devserver start logic to remove duplicated codes in next/swc.

### CODE REVIEW

File: crates/next-dev/src/main.rs [hunk 1]
```diff
@@ -101,32 +102,24 @@ fn default_host() -> IpAddr {
...
 
-#[cfg(not(feature = "cli"))]
-fn main() -> Result<()> {
-    unimplemented!("Cannot run binary without CLI feature enabled");
-}
-
-#[tokio::main]
-#[cfg(feature = "cli")]
-async fn main() -> Result<()> {
+pub async fn start_server(options: &DevServerOptions) -> Result<()> {
```
Comment (ForsakenHarmony) [inline crates/next-dev/src/main.rs:105]: I think this should be moved to `lib.rs`, `main.rs` should be just for the binary build, nothing else

**Review by ForsakenHarmony — CHANGES_REQUESTED** [review]

File: crates/next-dev/src/main.rs (hunk 1 above)
Comment (kwonoj) [reply crates/next-dev/src/main.rs:105]: yeah good point.

File: crates/next-dev/src/main.rs [hunk 2]
```diff
@@ -1,105 +1,9 @@
...
-fn default_port() -> u16 {
-    3000
-}
-
-#[cfg(feature = "serializable")]
-fn default_host() -> IpAddr {
-    IpAddr::V4(std::net::Ipv4Addr::new(0, 0, 0, 0))
-}
 
 #[cfg(not(feature = "cli"))]
```
Comment (ForsakenHarmony) [inline crates/next-dev/src/main.rs:8]: Do we still need this feature?

**Review by ForsakenHarmony — APPROVED** [review]

File: crates/next-dev/src/main.rs (hunk 2 above)
Comment (kwonoj) [reply crates/next-dev/src/main.rs:8]: This is I added newly, for the case if we want to create standalone binary uses clap (next.js napi bindings doesn't need). I can't say for sure how we'll deal with standalone binary yet, so preserved for the safety. afaik in a long run we want to have native standalone binary without bindings?

**Pull request merged**